src/MediaWiki.ts

```diff
@@ -215,6 +220,23 @@ class MediaWiki {
     return this.#hasRestApi
   }
 
+  public async hasActionParseApi(downloader: Downloader): Promise<boolean> {
+    if (this.#hasActionParseApi === null) {
+      for (const skin of ['vector-2022', 'vector']) {
+        this.actionParseUrlDirector = new ActionParseURLDirector(this.actionApiUrl.href, skin)
+        const checkUrl = this.actionParseUrlDirector.buildArticleURL(this.apiCheckArticleId)
+        this.#hasActionParseApi = await checkApiAvailability(downloader, checkUrl)
+        logger.log(`Checked for ActionParseApi with skin ${skin} at ${checkUrl} -- result is: ${this.#hasActionParseApi}`)
+        if (this.#hasActionParseApi) {
+          this.skin = skin
+          break
+        }
+      }
+      return this.#hasActionParseApi
```

Comment:
nit: no need to return here, you're returning it outside the if.

Reply:
Indeed, silly copy-paste. Fixed all occurrences (all checks for APIs)